### PR TITLE
Feat/tooltip guide link

### LINKED ISSUE #4076: Tooltip › Guide
The idea: Link from relevant tooltips to the respective article in Suite Guide. 🎉   

**Design:** [Zeplin
](https://zpl.io/bAKvAMx )
**Animation:** [Figma prototype](https://www.figma.com/proto/MmQtIPoIggyJkBm7vMUHpI/Neue-Desktop?page-id=0%3A1&node-id=7093%3A104058&viewport=-4441%2C66818%2C1.1297415494918823&scaling=min-zoom&starting-point-node-id=7093%3A104058&show-proto-sidebar=1)

### PR DESCRIPTION
**Pull request by dimasivan**
fix #4076 
@matejzak this PR solves guide opening from tooltip links (currently for Locktime and RBF both from send form view tooltips )

### CODE REVIEW

File: packages/suite/src/views/wallet/send/components/Options/components/BitcoinOptions/index.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 import styled from 'styled-components';
+import { darken } from 'polished';
 import { Translation } from '@suite-components';
 import { OnOffSwitcher } from '@wallet-components';
 import { Button, Tooltip } from '@trezor/components';
 import { useSendFormContext } from '@wallet-hooks';
 import { isEnabled as isFeatureEnabled } from '@suite-utils/features';
+import { useActions } from '@suite-hooks';
+import * as guideActions from '@suite-actions/guideActions';
+import { Node } from '@trezor/suite-data/src/guide/parser';
```
Comment (tomasklim) [inline packages/suite/src/views/wallet/send/components/Options/components/BitcoinOptions/index.tsx]:
```suggestion
import type { Node } from '@suite-types/guide';
```

Guide types from parser are reexported in `guide.tsx` file. (Note: We should change `guide.tsx` to `guide.ts`)

File: packages/suite/src/views/wallet/send/components/Options/components/BitcoinOptions/index.tsx [hunk 2]
```diff
@@ -55,6 +70,25 @@ const BitcoinOptions = () => {
...
+
+    const tooltipOpensGuide = (id: string) => {
+        const node: Node = {
+            type: 'page',
+            id: '',
+            locales: [''],
+            title: {
+                '': 'string',
+            },
+        };
```
Comment (tomasklim) [inline packages/suite/src/views/wallet/send/components/Options/components/BitcoinOptions/index.tsx]: 6a136f59fe7753e23471c59b93a6c4cb7674ede5

**Comment by dimasivan** [comment]
lets discuss guide usability when modal in open
image

File: packages/suite/src/actions/suite/analyticsActions.ts [hunk 3]
```diff
@@ -262,6 +262,12 @@ export type AnalyticsEvent =
               type: 'bug' | 'suggestion';
           };
       }
+    | {
+          type: 'guide/tooltip';
```
Comment (tomasklim) [inline packages/suite/src/actions/suite/analyticsActions.ts]: Sorry, I know that it is my wording 😀, but I would modify it to something more obvious. This sounds like some tooltip click inside the guide panel. If you have better wording, feel free to use it.

```suggestion
          type: 'guide/tooltip-link/navigation';
```

Please bump analytics version and add it to changelog. (I will update analytics docs with description what to do when you need to add new event)

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { TrezorDevice, AcquiredDevice, InjectedModalApplicationProps } from '@su
 import { useSelector, useActions } from '@suite-hooks';
 import * as routerActions from '@suite-actions/routerActions';
 import * as suiteActions from '@suite-actions/suiteActions';
+import { OpenGuideFromTooltip } from '@suite/views/guide';
```
Comment (tomasklim) [inline packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx]:
```suggestion
import { OpenGuideFromTooltip } from '@suite-views/guide';
```
Edit everywhere please

File: packages/suite/src/hooks/suite/useGuide.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { useActions, useSelector, useAnalytics } from '@suite-hooks';
+import * as guideActions from '@suite-actions/guideActions';
+import { getNodeById } from '@suite/utils/suite/guide';
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useGuide.ts]:
```suggestion
import { getNodeById } from '@suite-utils/guide';
```

File: packages/suite/src/hooks/suite/useGuide.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        openNode: guideActions.openNode,
+    });
+
+    const { indexNode } = useSelector(state => ({
+        indexNode: state.guide.indexNode,
+    }));
+
+    const openNodeById = (id: string) => {
+        if (!indexNode) {
+            console.error(`Index node was not found.`);
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useGuide.ts]: To make it clear in the logs.
```suggestion
            console.error(`Guide index node was not found.`);
```

File: packages/suite/src/hooks/suite/useGuide.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            return;
+        }
+        const node = getNodeById(id, indexNode);
+        if (!node) {
+            console.error(`Node with id: ${id} was not found.`);
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useGuide.ts]:
```suggestion
            console.error(`Guide node with id: ${id} was not found.`);
```

**Review by tomasklim — CHANGES_REQUESTED** [review]
```
<OpenGuideFromTooltip id="/privacy/remember-and-eject.md" />
```

I think that **id** prop is a place where it is easy to make a mistake (mistyping or modified structure in GB). Let's discuss it and find a nice solution on how to avoid future problems. @pingpwned @matejkriz

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx (hunk 4 above)
Comment (dimasivan) [reply packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx]: 90b9ca6cb461e549a88b4b1006dbaa18ad07f287

File: docs/misc/analytics.md [hunk 8]
```diff
@@ -87,9 +87,12 @@ Breaking change should bump major version. Any other change bumps minor version.
...
 2. **Option**: Get access to Keboola
 3. **Option**: Create a modified build of app with an analytics server URL pointing to your server
-4. **Option**: Edit NAT to resolve requests to `https://data.trezor.io/suite/log/web/stable.log` to your local server 
-   
+4. **Option**: Edit NAT to resolve requests to `https://data.trezor.io/suite/log/web/stable.log` to your local server
+
 ## Changelog
+### 1.12
+Added:
+- guide/tooltip-link/navigation
```
Comment (tomasklim) [inline docs/misc/analytics.md:124]:
```suggestion
Added:
- guide/tooltip-link/navigation
  - id: string

```

File: docs/misc/analytics.md [hunk 9]
```diff
@@ -19,13 +19,13 @@ Data from **production** builds (codesign branch) are sent to:
```
Comment (tomasklim) [inline docs/misc/analytics.md:19]: Please remove the formatting changes in this file and let's do it in a separate branch. For some reason, we have deactivated formatting of markdown files in docs folder in `.prettierignore`. I will address this issue in a separate branch.

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: docs/misc/analytics.md (hunk 9 above)
Comment (matejkriz) [reply docs/misc/analytics.md:19]: agree with auto-formatting everywhere in separate issue, but I think it's ok to keep those changes here... but no strong opinion

**Review by matejkriz — COMMENTED** [review]
> ```
> <OpenGuideFromTooltip id="/privacy/remember-and-eject.md" />
> ```
> 
> 
>      
>   
> 
> I think that **id** prop is a place where it is easy to make a mistake (mistyping or modified structure in GB). Let's discuss it and find a nice solution on how to avoid future problems. @pingpwned @matejkriz

Maybe we could get it somehow from `@trezor/suite-data/files/guide/index.json` file?

**Comment by tomasklim** [comment]
> > ```
> > <OpenGuideFromTooltip id="/privacy/remember-and-eject.md" />
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think that **id** prop is a place where it is easy to make a mistake (mistyping or modified structure in GB). Let's discuss it and find a nice solution on how to avoid future problems. @pingpwned @matejkriz
> 
> Maybe we could get it somehow from `@trezor/suite-data/files/guide/index.json` file?

Added to #3992

File: docs/misc/analytics.md (hunk 9 above)
Comment (dimasivan) [reply docs/misc/analytics.md:19]: Its hard to remove formatting only and keep changes I want to keep as it is automatic formatting in Atom :/
But we can format the rest of the code in separate issue for sure.

Comment (tomasklim) [reply docs/misc/analytics.md:19]: Visual studio code offers e.g. this Screenshot 2021-07-26 at 12 10 54

Ok resolving

**Review by tomasklim — COMMENTED** [review]
So far LGTM.

Ping me when animations are done.

**Comment by pingpwned** [comment]
fix #4076

File: packages/components/src/components/Tooltip/index.tsx [hunk 10]
```diff
@@ -122,11 +188,26 @@ const Tooltip = ({
...
                         </BoxRich>
                     ) : (
                         <BoxDefault maxWidth={maxWidth} tabIndex={-1} style={spring} {...attrs}>
-                            {content}
+                            {title && <StyledTooltipTitle>{title}</StyledTooltipTitle>}
+                            {openGuide && (
+                                <OpenGuideInner>
+                                    <StyledText>
+                                        <Fade direction="right" duration={300}>
+                                            Learn
```
Comment (matejkriz) [inline packages/components/src/components/Tooltip/index.tsx]: Has to be marked for translation.

File: packages/components/src/components/Tooltip/index.tsx [hunk 11]
```diff
@@ -5,22 +5,67 @@ import Tippy, { TippyProps } from '@tippyjs/react/headless';
 import { Instance, Props as TProps } from 'tippy.js';
 import { transparentize } from 'polished';
 import { Link } from '../typography/Link';
+import { Icon } from '../Icon';
+import { Fade } from 'react-awesome-reveal';
```
Comment (matejkriz) [inline packages/components/src/components/Tooltip/index.tsx]: `react-awesome-reveal` is dependency of `suite-web-landing` only. Not sure if it's useful to introduce it in components package.

I feel like we have too many animation libraries used :-D ( react-spring, framer-motion, css animations, react-awesome-reveal, ...)

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx [hunk 12]
```diff
@@ -129,6 +130,8 @@ const DeviceItem = ({ device, instances, closeModalApp, backgroundRoute }: Props
     const [isExpanded, setIsExpanded] = useState(true);
     const [animateArrow, setAnimateArrow] = useState(false);
 
+    const { openNodeById } = useGuide();
```
Comment (matejkriz) [inline packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx]: what about to move the hook inside OpenGuideFromTooltip component, so it has only `id` as prop and this hook  is used on one place only?

**Comment by matejkriz** [comment]
> lets discuss guide usability when modal in open

yeah, the design is broken :-( 
image

File: packages/components/src/components/Tooltip/index.tsx (hunk 10 above)
Comment (dimasivan) [reply packages/components/src/components/Tooltip/index.tsx]: 67dcf05

File: packages/components/src/components/Tooltip/index.tsx (hunk 11 above)
Comment (dimasivan) [reply packages/components/src/components/Tooltip/index.tsx]: 67dcf05

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx (hunk 12 above)
Comment (dimasivan) [reply packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx]: 67dcf05

File: packages/components/src/components/Tooltip/index.tsx (hunk 11 above)
Comment (matejkriz) [reply packages/components/src/components/Tooltip/index.tsx]: how is this addressed in the fixup? the usage of react-awesome-reveal was just moved from packages/components to packages/suite, right? So it should be defined as dependency in `packages/suite/package.json`?

Comment (dimasivan) [reply packages/components/src/components/Tooltip/index.tsx]: yes the usage was moved. Shall I `yarn add react-awesome-reveal` also in packages/suite?

Comment (matejkriz) [reply packages/components/src/components/Tooltip/index.tsx]: I think so.

File: packages/components/src/components/Tooltip/index.tsx [hunk 13]
```diff
@@ -5,22 +5,48 @@ import Tippy, { TippyProps } from '@tippyjs/react/headless';
...
     color: ${props => props.theme.TYPE_WHITE};
     font-weight: ${FONT_WEIGHT.MEDIUM};
-    border-radius: 5px;
+    border-radius: 10px;
     font-size: ${FONT_SIZE.TINY};
     text-align: left;
     box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
     max-width: ${props => props.maxWidth}px;
+    &:hover ${OpenGuideInner} {
+        width: 61px;
```
Comment (matejkriz) [inline packages/components/src/components/Tooltip/index.tsx]: this width depends on the text, right? Have you tried some more flexible solution? We will introduce another language very soon...

Comment (dimasivan) [reply packages/components/src/components/Tooltip/index.tsx]: sure, fixed in 1a9e24e28b84e41ea5232603340c00fc50714bbe

File: packages/components/src/components/Tooltip/index.tsx (hunk 11 above)
Comment (dimasivan) [reply packages/components/src/components/Tooltip/index.tsx]: added in 1a9e24e28b84e41ea5232603340c00fc50714bbe

**Review by matejkriz — APPROVED** [review]

**Pull request closed without merging**